pypicloud/storage/files.py

```diff
@@ -26,20 +29,42 @@ def configure(cls, settings):
         kwargs['directory'] = directory
         return kwargs
 
-    def get_path(self, package):
-        """ Get the fully-qualified file path for a package """
+    def get_path(self, package, metadata=False):
+        """
+        Get the fully-qualified file path for a package and its metadata file.
+        """
+        if metadata:
+            filename = self.METADATA_FILE
```

Comment:
This is going to use the same metadata file for all packages of the same name and version. I think in most cases it'll be fine, but it feels weird that the behavior will be different from S3. Also there are some edge cases, such as:
1) Upload package-1.0.tar.gz
2) Upload package-1.0.whl
3) Delete package-1.0.whl

Now there is no metadata file for the remaining package

Changing this will also require changes in the `list()` method too, though